packages/atomic/src/components/commerce/atomic-commerce-product-list/atomic-commerce-product-list.tsx

```diff
@@ -193,12 +194,48 @@ export class AtomicCommerceProductList
     );
   }
 
-  private getPropsForAtomicProduct(product: Product) {
+  private logWarning(message: string) {
+    this.bindings.engine.logger.warn(message);
+  }
+
+  private getInteractiveProduct(product: Product, index: number) {
+    let parentController: ProductListing | Search;
+    if (this.bindings.interfaceElement.type === 'product-listing') {
+      parentController = buildProductListing(this.bindings.engine);
+    } else {
+      parentController = buildSearch(this.bindings.engine);
+    }
+
+    const {name, price, productId, warning} =
+      ProductTemplatesHelpers.getRequiredProductPropertiesForAnalytics(product);
+
+    const pagination = parentController.pagination();
+    const position =
+      pagination.state.page * pagination.state.pageSize + index + 1;
+
+    const controller =
+      name === undefined || price === undefined || productId === undefined
+        ? undefined
```

Comment:
I think it's preferable to completely avoir building the interactive product controller when some required properties are missing. This is why I made it an optional prop on the atomic-product component.

Another option would be to initialize the controller regardless, and use placeholder values.